src/test/groovy/com/bmuschko/gradle/docker/internal/DefaultDockerConfigResolverTest.groovy

```diff
@@ -0,0 +1,115 @@
+package com.bmuschko.gradle.docker.internal
+
+import spock.lang.Specification
+import spock.lang.TempDir
+import spock.lang.Unroll
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+class DefaultDockerConfigResolverTest extends Specification {
+
+    private static String fallbackDockerUrl = 'tcp://127.0.0.1:2375'
+
+    @TempDir
+    Path temporaryFolder
+
+    void setup() throws IOException {
+        SystemConfig.clear()
+
+        // override system envs with null values, to avoid flaky side effects
+        SystemConfig.setEnv('DOCKER_HOST', null)
```

Comment:
The relevant test methods should use the Spock annotation `@RestoreSystemProperties` which will restore the original values after a test.

Reply:
My bet. You are right. Looks like you remove the implementation.